test/src/unit/modules/ProjectList.test.js

```diff
@@ -0,0 +1,297 @@
+            const project = new Project({ name: 'newdummyproject' }, global.codewind.CODEWIND_WORKSPACE);
+            projectList.addProject(project);
+            const exists = projectList.projectNameExists('newdummyproject');
+            exists.should.be.true;
+        });
+        it('Returns false as ProjectList is empty', () => {
+            const projectList = new ProjectList();
+            const exists = projectList.projectNameExists('random');
+            exists.should.be.false;
+        });
+        it('Returns false as project name does not exist', () => {
+            const projectList = new ProjectList();
+            const project = new Project({ name: 'newdummyproject' }, global.codewind.CODEWIND_WORKSPACE);
+            projectList.addProject(project);
+            const exists = projectList.projectNameExists('random');
+            exists.should.be.false;
+        });
+    });
+    describe('removeProject(id)', () => {
+        it('Creates a new Project and removes it from the ProjectList', () => {
+            const projectList = new ProjectList();
+            const project = new Project({ name: 'newdummyproject' }, global.codewind.CODEWIND_WORKSPACE);
+            projectList.addProject(project);
+            const list = projectList.list;
+            list.should.have.property(project.projectID);
+
+            projectList.removeProject(project.projectID);
+            list.should.eql({});
+            Object.keys(list).length.should.equal(0);
+            list.should.not.have.property(project.projectID);
+        });
+        it('Throws an error as the project does not exist', () => {
+            const projectList = new ProjectList();
+            const id = '123';
+            const error = new ProjectListError('NOT_FOUND', id);
+            assert.throws(() => projectList.removeProject(id), error);
+        });
+    });
+    describe('retrieveProject(id)', () => {
+        it('Retrieves a project that exists', () => {
+            const projectList = new ProjectList();
+            const project = new Project({ name: 'newdummyproject' }, global.codewind.CODEWIND_WORKSPACE);
+            projectList.addProject(project);
+            const list = projectList.list;
+            list.should.have.property(project.projectID);
+
+            const retrievedProject = projectList.retrieveProject(project.projectID);
+            retrievedProject.should.deep.equal(project);
+        });
+        it('Receives undefined as project does not exist', () => {
+            const projectList = new ProjectList();
+            const id = '123';
+            const retrievedProject = projectList.retrieveProject(id);
+            /*eslint no-undef: "error"*/
+            // eslint-disable-next-line no-undefined
+            assert.equal(retrievedProject, undefined);
+        });
+    });
+    describe('updateProject(updatedProject)', () => {
+        it('Errors as project does not have a projectID', () => {
```

Comment:
you could potentially split these and others up by success and failure describe blocks (within each function describe), up to you though.